components/tidb_query_normal_expr/src/builtin_time.rs

```diff
@@ -557,6 +557,32 @@ impl ScalarFunc {
         let time = Time::from_days(ctx, days)?;
         Ok(Some(Cow::Owned(time)))
     }
+
+    #[inline]
+    pub fn make_date<'a>(
+        &self,
+        ctx: &mut EvalContext,
+        row: &[Datum],
+    ) -> Result<Option<Cow<'a, Time>>> {
+        let mut year = try_opt!(self.children[0].eval_int(ctx, row)) as u32;
+        let mut days = try_opt!(self.children[1].eval_int(ctx, row)) as u32;
+        if year > 9999 {
+            year = 10000 // for returning error
```

Comment:
Why not just return an error here? 😆

Reply:
Good idea :) I've added returning error.